src/main/java/net/minecraftforge/common/util/BlockSnapshot.java

```diff
@@ -135,18 +120,28 @@ public static BlockSnapshot readFromNBT(NBTTagCompound tag)
                 nbt);
     }
 
+    private static NBTTagCompound getTileNBT(TileEntity te)
+    {
+        if (te == null) return null;
+        NBTTagCompound nbt = new NBTTagCompound();
+        te.writeToNBT(nbt);
+        return nbt;
+    }
+
     public IBlockState getCurrentBlock()
     {
         return getWorld().getBlockState(getPos());
     }
 
     public World getWorld()
     {
-        if (this.world == null)
+        World world = this.world != null ? this.world.get() : null;
```

Comment:
It'd have to be `if (this.world == null || this.world.get() == null)` in that case, the ternary isn't great but we don't have `?.` as an operator in Java.